eclipse-collections/src/main/java/org/eclipse/collections/impl/set/sorted/immutable/AbstractImmutableSortedSet.java

```diff
@@ -488,13 +489,10 @@ public boolean isProperSubsetOf(SetIterable<? extends T> candidateSuperset)
     @Override
     public ParallelSortedSetIterable<T> asParallel(ExecutorService executorService, int batchSize)
     {
-        if (executorService == null)
-        {
-            throw new NullPointerException();
-        }
+        Objects.requireNonNull(executorService);
         if (batchSize < 1)
         {
-            throw new IllegalArgumentException();
+            throw new IllegalArgumentException("batchSize must be greater than zero, but was: " + batchSize);
         }
         return new NonParallelSortedSetIterable<>(this);
```

Comment:
Is this overridden? I can't think of a good way to make this efficient with the building blocks we have.

Reply:
Yep, just in `ImmutableTreeSet` but not all the methods are implemented with parallelization